src/index.ts

```diff
@@ -3973,7 +3973,10 @@ export default class Elysia<
 		options: string | number | Partial<Serve>,
 		callback?: ListenCallback
 	) => {
-		if (!Bun) throw new Error('Bun to run')
+		if (typeof Bun === 'undefined')
+			throw new Error(
+				'.listen() is designed to run on Bun only. If you are running Elysia in other environment please use a dedicated plugin or export the handler via Elysia.fetch'
```

Comment:
"... designed to only run in Bun" sounds a bit better  instead of "...designed to run on Bun" imo